src/main/java/pl/foodRecipe/domain/category/CategoryRepository.java

```diff
@@ -0,0 +1,9 @@
+package pl.foodRecipe.domain.category;
+
+import org.springframework.data.repository.CrudRepository;
+
+import java.util.Optional;
+
+public interface CategoryRepository extends CrudRepository<Category, Long> {
+    Optional<Category> findByNameIgnoreCase(String name);
```

Comment:
Zastanowiłbym się czy nie lepiej było by wyszukiwanie po części nazwy. Jeśli nie wiem czego szukam (dokładna nazwa) to wyszukiwałbym po nie pełnym słowie